src/Messages/OWSIdentityManager.m

```diff
@@ -186,24 +214,33 @@ - (void)setVerificationState:(OWSVerificationState)verificationState
         // it for the first time.
         [self saveRemoteIdentity:identityKey recipientId:recipientId];
 
-        OWSRecipientIdentity *identity = [OWSRecipientIdentity fetchObjectWithUniqueID:recipientId];
+        OWSRecipientIdentity *recipientIdentity = [OWSRecipientIdentity fetchObjectWithUniqueID:recipientId];
 
-        if (identity == nil) {
+        if (recipientIdentity == nil) {
             OWSFail(@"Missing expected identity: %@", recipientId);
             return;
         }
 
-        if (identity.verificationState == verificationState) {
+        if (recipientIdentity.verificationState == verificationState) {
             return;
         }
 
         DDLogInfo(@"%@ setVerificationState: %@ (%@ -> %@)",
             self.tag,
             recipientId,
-            OWSVerificationStateToString(identity.verificationState),
+            OWSVerificationStateToString(recipientIdentity.verificationState),
             OWSVerificationStateToString(verificationState));
 
-        [identity updateWithVerificationState:verificationState];
+        [recipientIdentity updateWithVerificationState:verificationState];
+
+        if (sendSyncMessage) {
+            [self enqueueSyncMessageForVerificationState:verificationState
+                                             identityKey:identityKey
+                                             recipientId:recipientId];
+        } else {
+            // Cancel any pending verification state sync messages for this recipient.
+            [self clearSyncMessageForRecipientId:recipientId];
```

Comment:
Just typing this out to make sure I understand...

...so **receiving** a verification sync will cancel an in-progress sending sync. 

Presumably this is most relevant when bringing a device back online, after being offline for a while, in which case any locally queued outgoing verification sync message is probably less relevant than any that have happened since on the secondary device.

So, I *think* this will usually be right. I don't think we *can* be right in every case, which is why we have an audit trail of info messages. 👍